### ISSUE TITLE
Question: How do I avoid overfitting with ELMo finetuning

### ISSUE THREAD

**Issue by rosequ** [allenai/bilm-tf#138]
I finetune ELMo on our domain data which is 41,581,796 sentences and reach a train perplexity around 50 but validation perplexity is still high (around 105). How do I avoid overfitting while finetuning ELMo model? 
I don't have enough resources to train our own ELMo models.

**Comment by matt-peters** [comment]
This is a very large dataset and would be difficult to overfit.  If you are concerned about overfitting, then I suggest evaluating on a held out set every epoch to determine when to stop tuning.

Also note that the training perplexity isn't directly comparable to test perplexity due to the sampled softmax, and will typically be much lower (during training, perplexity is computed with only a small set of negative samples instead of using the full vocabulary). To check for overfitting, you could take a small piece of your training set and check perplexity of that in testing mode (without sampling or dropout).

**Comment by rosequ** [comment]
I see. That helps! You have been by far the most reachable NLP researcher! Thanks a lot for the record time quick reply!
